### PR TITLE
Simplify counter and add Gauge, Histogram, Summary and reorganise code...

### PR DESCRIPTION
**Pull request by Krule**
1. Counter is supposed to be increased by a whole number only as far as I can tell by reading Ruby gem.
2. No need to check if it's greater than 0 when we can use `UInt`.
3. Is reset really needed?
4. Instead of `#get` a `#value` could be used and this makes it possible to define it as getter
5. Document 😄

Also... what do you think about moving `Metric` and `Registry` and stuff outside `Client` scope and leaving `Client` to do client stuff (like sending data over the network)?

---
Update... added bunch of stuff. Can't sleep as I slept too much today and had a lot of coffee later.

Also not sure about `#increment` and `#decrement` in `Gauge`. These look like convenience methods in Ruby gem to turn `Gauge` to bi-directional counter. Do we want this?

### CODE REVIEW

**Comment by mhenrixon** [comment]
Lets move! 👍

**Comment by Krule** [comment]
Compared to Ruby gem what I added is still pretty elementary. Missing labels supports as well.

**Comment by Krule** [comment]
I should sleep. 👋

**Comment by tristanholl** [comment]
I like the idea of moving the different metrics out of the client context, not really sure why the ruby implementation assumes they are part of it.

**Comment by tristanholl** [comment]
Isn't the Gauge missing the option to just set a value?

**Comment by Krule** [comment]
> Isn't the Gauge missing the option to just set a value?

That would be something like idiomatic Ruby where setters are like `gauge.value = 42` 😄

File: src/metrics_cr/metrics/gauge.cr [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+module MetricsCr
+  module Metrics
+    class Gauge < Metric
+      property value : Float64 = 0
+
+      # Decrements the gauge by provided value.
+      #
+      # ```
+      # gauge = MetricsCr::Metrics::Gauge.new(name: "test")
+      # gauge.value = 3.14159    # => Nil
```
Comment (Krule) [inline src/metrics_cr/metrics/gauge.cr]: @tristanholl Check this out in regards to setter.

**Comment by tristanholl** [comment]
> > Isn't the Gauge missing the option to just set a value?
> 
> That would be something like idiomatic Ruby where setters are like `gauge.value = 42` 😄

Totally missed the property 🤦‍♂

**Review by tristanholl — APPROVED** [review]
Looks nice, LGTM and continue in another PR 👍

**Pull request merged**